tests/references/collections.py

```diff
@@ -64,153 +88,831 @@
+                    },
+                ]
+            }
+        },
+    },
+    {
+        "desc": cleandoc("""Tokens with multiple values for same trait type."""),
+        "token_type": "non-fungible",
+        "tokens": {
+            1: {
+                "attributes": [
+                    {"name": "weapon", "value": "axe"},
+                    {"name": "weapon", "value": "kitana"},
+                    {"name": "Animated ", "value": "No "},
+                ]
+            }
+        },
+        "_expected": {
+            1: {
+                "attributes": [
+                    {
+                        "name": "animated",
+                        "value": "no",
+                        "display_type": "string",
+                    },
+                    {
+                        "name": "weapon",
+                        "value": "axe",
+                        "display_type": "string",
+                    },
+                    {
+                        "name": "weapon",
+                        "value": "kitana",
+                        "display_type": "string",
+                    },
+                    {
+                        "name": "openrarity.trait_count",
+                        "value": 3,
                         "display_type": "string",
                     },
                 ]
+            }
+        },
+    },
+]
+
+
+TEST_ATTRIBUTE_STATISTICS_PARAMS = [
+    {
+        "desc": cleandoc("""Tokens with string traits."""),
+        "token_type": "non-fungible",
+        "tokens": {
+            1: {
+                "attributes": [
+                    {"name": "weapon", "value": "axe"},
+                    {"name": "animated", "value": "no"},
+                    {"name": "eyewear", "value": "monocle"},
+                ]
+            },
+            2: {
+                "attributes": [
+                    {"name": "weapon", "value": "kitana"},
+                    {"name": "animated", "value": "no"},
+                ]
+            },
+        },
+        "_expected": {
+            "string": [
+                {
+                    "name": "eyewear",
+                    "value": "monocle",
+                    "attribute.token_count": 1,
+                    "attribute.supply": 1,
+                    "metric.probability": 0.5,
+                    "metric.information": 1.0,
+                },
+                {
+                    "name": "weapon",
+                    "value": "axe",
+                    "attribute.token_count": 1,
+                    "attribute.supply": 1,
+                    "metric.probability": 0.5,
+                    "metric.information": 1.0,
+                },
+                {
+                    "name": "animated",
+                    "value": "no",
+                    "attribute.token_count": 2,
+                    "attribute.supply": 2,
+                    "metric.probability": 1.0,
+                    "metric.information": -0.0,
```

Comment:
Any idea what the negative is about? Does the library output like that?

Reply:
The library outputs this, `-log2(attr["attribute.supply"] / total)` will return -0.0 when attribute supply equals total supply.

Reply:
Ah right right that makes sense. And besides `-0.0 == 0.0` is `True` so nbd.